geoengine/workflow.py

```diff
@@ -268,9 +270,9 @@ def plot_chart(self, bbox: QueryRectangle, timeout: int = 3600) -> VegaLite:
 
         check_response_for_error(response)
 
-        response = response.json()
+        response_json: Dict[str, Any] = response.json()
```

Comment:
Maybe we can be more concrete than any (cf. above)

Reply:
I think this is something for a future mypy version. There hasn't been support for recursive typedefs in mypy and it's been integrated into the mainbranch just three weeks ago [Support recursive types #731](https://github.com/python/mypy/issues/731). 
I'm not sure if that would be a solution for json though, since its flexible nature.

Reply:
The only answer isn't even selected as an answer to that problem.

So what would you do then?

Basically, this type should at least reflect JSON in the supertype, i.e.,  A -> str -> Union[str, number, bool, A] or something like this. Cf. JavaScript code: https://github.com/geo-engine/geoengine-ui/blob/30930bf18ca656feb981a88d299dd0b31942f942/projects/wave-core-new/src/lib/backend/backend.model.ts#L246

The subtype should have a more concrete typing. Maybe you can use `TypedDict` for that?

Reply:
This [0223d46](https://github.com/geo-engine/geoengine-python/pull/70/commits/0223d46535d65dbf741c9d13a8c52b5a483194a0) is the closest thing I could find as a working solution. 
Downside: It needs an assertion here and there to hint mypy in the right direction.

Using `TypedDict` would mean that we need to know the key names explicitly, if I'm not mistaken. 
I think this is not applicable for the response.json() scenario.

Reply:
Well, it is. We know the returned JSON structure, otherwise we could not extract fields.

Reply:
> * JsonType = Union[Dict[str, Any], List[Any], int, str, float, bool, Type[None]]
>   Can't we self-referentiate? Dict[str, JsonType] ?

Unfortunately not, this is the big problem with defining json in python at the moment.
See this [mypy-playground](https://mypy-play.net/?mypy=latest&python=3.10).

It **might** become possible in the next mypy release though.

Reply:
> * Well, it is. We know the returned JSON structure, otherwise we could not extract fields.

Ok, so for using a `TypedDict` I have implemented all the variations necessary to replace things using `Any`. Is this closer to an acceptable solution?

Reply:
Could you add a comment like
```
# TODO: Define as recursive type when supported in mypy: https://github.com/python/mypy/issues/731
```
?

Reply:
Added with [315545c](https://github.com/geo-engine/geoengine-python/pull/70/commits/315545c0a3190709c568166c3df0c48134db3ef9)